web/docs/auth/auth-hooks.md

```diff
@@ -0,0 +1,293 @@
+
+
+### `onBeforeSignup` hook
+
+
+This hook is called before the user signs up. It is an async function that Wasp awaits before proceeding with the signup.
+
+
+:::info
+
+It can be useful if you want to reject a user based on some criteria before they sign up.
+:::
+
+Works with <EmailPill /> <UsernameAndPasswordPill /> <GithubPill /> <GooglePill /> <KeycloakPill />
+
+Its **input** is an `args` object with the following properties:
+
+- `providerId: { providerName: string, providerUserId: string }` - The provider ID of the user
+- `prisma: PrismaClient` - Prisma client instance
+- `req: Request` - Express request object
+- `hookName: string` - The name of the hook that is being called (in this case, it will be `'onBeforeSignup'`)
+
+The **return value** of this hook is ignored.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+app myApp {
+  ...
+  auth: {
+    ...
+    onBeforeSignup: import { onBeforeSignup } from "@src/auth/hooks",
+  },
+}
+```
+
+```js title="src/auth/hooks.js"
+export const onBeforeSignup = async ({ providerId, prisma, req, hookName }) => {
+  // Do something before signup
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+app myApp {
+  ...
+  auth: {
+    ...
+    onBeforeSignup: import { onBeforeSignup } from "@src/auth/hooks",
+  },
+}
+```
+
+```ts title="src/auth/hooks.ts"
+import type { OnBeforeSignupHookFn } from 'wasp/server/auth'
+
+export const onBeforeSignup: OnBeforeSignupHookFn = async ({ providerId, prisma, req, hookName }) => {
+  // Do something before signup
+}
+```
+
+</TabItem>
+</Tabs>
+
+
+### `onAfterSignup` hook
+
+
+This hook is called after the user signs up. It is an async function that Wasp awaits before proceeding with the signup.
+
+:::info
+
+It can be useful if you want to send the user a welcome email or perform some other action after the user signs up like syncing the user with a third-party service.
+:::
+
+Works with <EmailPill /> <UsernameAndPasswordPill /> <GithubPill /> <GooglePill /> <KeycloakPill />
+
+Its **input** is an `args` object with the following properties:
+- `providerId: { providerName: string, providerUserId: string }` - The provider ID of the user
+- `user: User` - The user object that was created
+- `prisma: PrismaClient` - Prisma client instance
+- `req: Request` - Express request object
+- `hookName: string` - The name of the hook that is being called (in this case, it will be `'onAfterSignup'`)
+
+The **return value** of this hook is ignored.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+app myApp {
+  ...
+  auth: {
+    ...
+    onAfterSignup: import { onAfterSignup } from "@src/auth/hooks",
+  },
+}
+```
+
+```js title="src/auth/hooks.js"
+export const onAfterSignup = async ({ providerId, user, prisma, req, hookName }) => {
+  // Do something after signup
```

Comment:
TODO: add a better example e.g. set a field on the user entity or send data to an external service.